src/client/PyCallbacks.h

```diff
@@ -49,17 +49,24 @@ class PyCallbacks : public IPyCallbacks {
 		py::dict mCGAReport;
 		std::wstring mCGAPrints;
 		std::vector<std::wstring> mCGAErrors;
+		py::dict mAttrVal;
 	};
 
 	std::vector<Model> mModels;
+	RuleFileInfoUPtr mRuleFileInfo;
 
 public:
-	PyCallbacks(const size_t initialShapeCount) {
+	PyCallbacks(const size_t initialShapeCount, RuleFileInfoUPtr& ruleFileInfo) {

```

Comment:
this fails to build with GCC, see branch build. ruleFileInfo needs to be an rvalue to be correctly "moved", i.e. `RuleFileInfoUPtr&& ruleFileInfo`

however, I would recommend a different approach and avoiding moving around the ownership of ruleInfo, see main review comment